isbn_updater/src/comp.php

```diff
@@ -0,0 +1,352 @@
+	$tokenURL = 'https://users.talis.com/oauth/tokens';
+	$content = "grant_type=client_credentials";
+
+	//*********GET DATE**********************
+
+	$date = date('Y-m-d\TH:i:s');
+	// $date1 = "2015-12-21T15:44:36";
+
+	//************GET_TOKEN***************
+
+	$ch = curl_init();
+
+	curl_setopt($ch, CURLOPT_URL, $tokenURL);
+	curl_setopt($ch, CURLOPT_POST, TRUE);
+	curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+	curl_setopt($ch, CURLOPT_USERPWD, "$clientID:$secret");
+	curl_setopt($ch, CURLOPT_POSTFIELDS, $content);
+
+	$return = curl_exec($ch);
+	$info = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+	if ($info !== 200){
+		echo "<p>ERROR: There was an error getting a token:</p><pre>" . var_export($return, true) . "</pre>";
+	} else {
+		echo "Got Token</br>";
+	}
+
+	curl_close($ch);
+
+	$jsontoken = json_decode($return);
+
+	if (!empty($jsontoken->access_token)){
+		$token = $jsontoken->access_token;
+	} else {
+		echo "<p>ERROR: Unable to get an access token</p>";
+		exit;
+	}
+
+	return $token;
+}
+
+/**
+ * isValidIsbn13
+ * https://en.wikipedia.org/wiki/ISBN#ISBN-13_check_digit_calculation
+ *
+ * @param  string $isbn
+ * @return boolean
+ */
+function isValidIsbn13($isbn)
+{
+    $check = 0;
+
+	// if it looks like an ISBN13
+	if (preg_match("/^97[89]\d{9}[\dxX]$/", $isbn)){
+		for ($i = 0; $i < 13; $i += 2) {
+			$check += (int)$isbn[$i];
+		}
+		
+		for ($i = 1; $i < 12; $i += 2) {
+			$check += 3 * $isbn[$i];
+		}
+		// and if the check digit is valid.
+		return (0 === ($check % 10)) ? true : false;
+	}
+
+	// else it can't be an ISBN13
+	return false;		
+}
+
+/**
+ * Update the resource if we need to
+ *
+ * @param  mixed $resource
+ * @param  mixed $old_isbn
+ * @param  mixed $new_isbn
+ * @return boolean|array False or the new ISBN13s to update
+ */
+function apply_update_logic($resource, $old_isbn, $new_isbn){
+	// if there are some isbn13s to check
+	if (!empty($resource->attributes->isbn13s)) {
+		// make a copy of the isbns to keep any additional ones (safest)
+		$output_isbn13s = $resource->attributes->isbn13s;
+		
+		// for each of the input ISBNs, see if it should be updated
+		foreach ($resource->attributes->isbn13s as $key => $value) {
+			if ($value == $old_isbn){
+				$output_isbn13s[$key] = $new_isbn;
+			}
+		}
+		
+		// if the input and output are not the same then changes were made.
+		if ($output_isbn13s !== $resource->attributes->isbn13s){
+			return $output_isbn13s;
+		}
+	}
+	// otherwise return false as there is nothing to do
+	return false;
+}
+
+/**
+ * Update the resource
+ *
+ * @param  string $shortCode tenant short code
+ * @param  string $resource_id the resource_id to update
+ * @param  string $TalisGUID The user making the update
+ * @param  string $token Your access token for the API
+ * @param  array $new_isbn13s 
+ * @param  mixed $myfile Log output to a file.
+ * @return void
+ */
+function updateResource($shortCode, $resource_id, $TalisGUID, $token, $new_isbn13s, $myfile) {
```

Comment:
```suggestion
function updateResource($shortCode, $resource_id, $TalisGUID, $token, array $new_isbn13s, $myfile) {
```

Suggestion: type-hint the `$new_isbn13s` array.  Elsewhere we are calling a function that will either return a new array or `false` if there are no changes to be made.  This type hint will guard against potential future bugs where false isn't checked for before passing it to this function.